semgrep-core/src/engine/Range_with_metavars.ml

```diff
@@ -37,11 +37,14 @@ let (range_to_pattern_match_adjusted : Rule.t -> t -> Pattern_match.t) =
  fun r range ->
   let m = range.origin in
   let rule_id = m.rule_id in
+  let languages = Xlang.to_langs r.Rule.languages in
   (* adjust the rule id *)
```

Comment:
Yeah, we might want to change the name `rule_id` at this point. Maybe `rule_info`?

Reply:
Maybe. I'm actually not sure why we don't just pass the whole rule around. When I last looked into this my guess was that it was initially actually just the rule ID, but then people (myself included) added more and more info that would be nice to have.